Flaky config flow Shelly test

### The problem

Sometimes I see this test is failing on unrelated changes and this requires restarting of the CI.

Example of failed job on dev branch: https://github.com/home-assistant/core/runs/2618862701

Test name: `tests/components/shelly/test_config_flow.py::test_form`

### What is version of Home Assistant Core has the issue?

dev

### What was the last working version of Home Assistant Core?

_No response_

### What type of installation are you running?

Home Assistant Core

### Integration causing the issue

shelly

### Link to integration documentation on our website

_No response_

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

```txt
==================================== ERRORS ====================================
_____________________ ERROR at setup of test_form[pyloop] ______________________
[gw1] linux -- Python 3.8.2 /__w/core/core/venv/bin/python

loop_factory = <class 'asyncio.unix_events._UnixDefaultEventLoopPolicy'>
fast = False, loop_debug = False

    @pytest.fixture
    def loop(loop_factory, fast, loop_debug):  # type: ignore
        """Return an instance of the event loop."""
        policy = loop_factory()
        asyncio.set_event_loop_policy(policy)
>       with loop_context(fast=fast) as _loop:

venv/lib/python3.8/site-packages/aiohttp/pytest_plugin.py:228: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
venv/lib/python3.8/site-packages/aiohttp/test_utils.py:493: in loop_context
    loop = setup_test_loop(loop_factory)
venv/lib/python3.8/site-packages/aiohttp/test_utils.py:514: in setup_test_loop
    asyncio.set_event_loop(loop)
/usr/local/lib/python3.8/asyncio/events.py:753: in set_event_loop
    get_event_loop_policy().set_event_loop(loop)
/usr/local/lib/python3.8/asyncio/unix_events.py:1365: in set_event_loop
    self._watcher.attach_loop(loop)
/usr/local/lib/python3.8/asyncio/unix_events.py:931: in attach_loop
    self._do_waitpid_all()
/usr/local/lib/python3.8/asyncio/unix_events.py:985: in _do_waitpid_all
    self._do_waitpid(pid)
/usr/local/lib/python3.8/asyncio/unix_events.py:1019: in _do_waitpid
    callback(pid, returncode, *args)
/usr/local/lib/python3.8/asyncio/unix_events.py:216: in _child_watcher_callback
    self.call_soon_threadsafe(transp._process_exited, returncode)
/usr/local/lib/python3.8/asyncio/base_events.py:764: in call_soon_threadsafe
    self._check_closed()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <_UnixSelectorEventLoop running=False closed=True debug=False>

    def _check_closed(self):
        if self._closed:
>           raise RuntimeError('Event loop is closed')
E           RuntimeError: Event loop is closed

/usr/local/lib/python3.8/asyncio/base_events.py:508: RuntimeError
---------------------------- Captured stderr setup -----------------------------
DEBUG:asyncio:Using selector: EpollSelector
------------------------------ Captured log setup ------------------------------
DEBUG    asyncio:selector_events.py:59 Using selector: EpollSelector
=========================== short test summary info ============================
ERROR tests/components/shelly/test_config_flow.py::test_form[pyloop] - Runtim...
```


### Additional information

_No response_

--- a/tests/components/shelly/conftest.py
+++ b/tests/components/shelly/conftest.py
@@ -66,7 +66,7 @@ MOCK_SHELLY = {
 @pytest.fixture(autouse=True)
 def mock_coap():
     """Mock out coap."""
-    with patch("homeassistant.components.shelly.get_coap_context"):
+    with patch("homeassistant.components.shelly.utils.get_coap_context"):
         yield
 
 

